tests/test_cli.py

```diff
@@ -0,0 +1,286 @@
+# to update, use: `COLUMNS=100000 poetry run mcstatus --help`
+EXPECTED_HELP_OUTPUT = """
+usage: mcstatus [-h] [--bedrock] address {ping,status,query,json} ...
+
+mcstatus provides an easy way to query Minecraft servers for any information they can expose. It provides three modes of access: query, status, ping and json.
+
+positional arguments:
+  address               The address of the server.
+
+options:
+  -h, --help            show this help message and exit
+  --bedrock             Specifies that 'address' is a Bedrock server (default: Java).
+
+commands:
+  Command to run, defaults to 'status'.
+
+  {ping,status,query,json}
+    ping                Ping server for latency.
+    status              Prints server status. Supported by all Minecraft servers that are version 1.7 or higher.
+    query               Prints detailed server information. Must be enabled in servers' server.properties file.
+    json                Prints server status and query in json. Supported by all Minecraft servers that are version 1.7 or higher.
+"""  # noqa: E501(line length)
+
+
+@contextlib.contextmanager
+def patch_stdout_stderr():
+    outpatch = patch("sys.stdout", new=io.StringIO())
+    errpatch = patch("sys.stderr", new=io.StringIO())
+    with outpatch as out, errpatch as err:
+        yield out, err
+
+
+@pytest.fixture
+def mock_network_requests():
+    with \
+        patch("mcstatus.server.JavaServer.lookup", return_value=JavaServer("example.com", port=25565)), \
+        patch("mcstatus.server.JavaServer.ping", return_value=0), \
+        patch("mcstatus.server.JavaServer.status", return_value=JavaStatusResponse.build(JAVA_RAW_RESPONSE)), \
+        patch("mcstatus.server.JavaServer.query", return_value=QueryResponse(*QUERY_RAW_RESPONSE)), \
+        patch("mcstatus.server.BedrockServer.lookup", return_value=BedrockServer("example.com", port=25565)), \
+        patch("mcstatus.server.BedrockServer.status", return_value=BedrockStatusResponse.build(BEDROCK_RAW_RESPONSE, latency=0)
+    ):  # fmt: skip # multiline with was added in Python 3.10
+        yield
+
+
+def normalise_help_output(s: str) -> str:
+    """
+    Normalises the output of `mcstatus --help`, to work around
+    some discrepancies between Python versions while still retaining
+    meaningful information for comparison.
+    """
+
+    elided = "[...]:"
+
+    s = s.strip()
+
+    # drop lines which end in ":". these argparse section headings vary between python versions.
+    return "\n".join(ln if not ln.endswith(":") else elided for ln in s.splitlines())
+
+
+# NOTE: for premature exits in argparse, we must catch SystemExit.
+# for ordinary exits in the CLI code, we can simply inspect the return value.
+
+
+def test_no_args():
+    with patch_stdout_stderr() as (out, _), pytest.raises(SystemExit) as exn:
+        main_under_test([])
+
+    assert out.getvalue() == ""
+    assert exn.value.code != 0
```

Comment:
We should probably also check that stderr isn't empty, or maybe even that it contains the help message, perhaps like it's done in `test_help`, i.e. `"usage: " in err.getvalue()`

Reply:
Done, thanks!